src/main/java/org/rumbledb/compiler/TranslationVisitor.java

```diff
@@ -847,9 +844,14 @@ public Void visitObjectConstructor(JsoniqParser.ObjectConstructorContext ctx) {
             List<Expression> keys = new ArrayList<>();
             List<Expression> values = new ArrayList<>();
             for (JsoniqParser.PairConstructorContext currentPair : ctx.pairConstructor()) {
-                this.visitPairConstructor(currentPair);
-                keys.add(this.currentExpression);
-                values.add(this.otherCurrentExpression);
+                if (currentPair.lhs != null) {
+                    this.visitExprSingle(currentPair.lhs);
+                    keys.add(this.currentExpression);
+                } else {
+                    keys.add(new StringLiteralExpression(currentPair.name.getText(), createMetadataFromContext(ctx)));
+                }
+                this.visitExprSingle(currentPair.rhs);
+                values.add(this.currentExpression);
             }
             this.currentExpression = new ObjectConstructorExpression(keys, values, createMetadataFromContext(ctx));
```

Comment:
I have a question here. The objectConstructorExpression seems to take lists of expressions for its keys and values it is always passed a singleton(this is the only place this class seems to be used). I was wondering if this is in your agenda for issues to address.

Reply:
I am not sure I understand your question. Could you elaborate? Thanks!

Reply:
I misinterpreted the code here last time I looked at it. On a second look, it looks good as it is.